### ISSUE TITLE
Exception: "Input string was not in a correct format."

### ISSUE THREAD

**Issue by andrasfuchs** [Genbox/WolframAlpha#11]
I used the Query method with the following parameter:
`WolframAlpha.Query("Plus[7, Times[7, x]]=0")`

The API request was successful, and the response was the following:

```
<?xml version='1.0' encoding='UTF-8'?>
<queryresult success='true'
    error='false'
    numpods='1'
    datatypes=''
    timedout=''
    timedoutpods=''
    timing='1.0230000000000001'
    parsetiming='0.613'
    parsetimedout='false'
    recalculate=''
    id='MSP1801h5c3fg92f0194960000253b8ice3ghfff5i'
    host='https://www5b.wolframalpha.com'
    server='31'
    related='https://www5b.wolframalpha.com/api/v1/relatedQueries.jsp?id=MSPa1811h5c3fg92f01949600001di79bh0edie2i4e5282741642789771482'
    version='2.6'>
 <pod title='Solution'
     scanner='Reduce'
     id='Solution'
     position='100'
     error='false'
     numsubpods='1'
     primary='true'>
  <subpod title=''>
   <img src='https://www5b.wolframalpha.com/Calculate/MSP/MSP1821h5c3fg92f01949600000edggfa59013habb?MSPStoreType=image/gif&amp;s=31'
       alt='x = -1'
       title='x = -1'
       width='550'
       height='22'
       type='Default'
       themes='1,2,3,4,5,6,7,8,9,10,11,12'
       colorinvertable='true' />
   <plaintext>x = -1</plaintext>
  </subpod>
  <expressiontypes count='1'>
   <expressiontype name='Default' />
  </expressiontypes>
  <states count='1'>
   <state name='Step-by-step solution'
       input='Solution__Step-by-step solution'
       stepbystep='true' />
  </states>
 </pod>
</queryresult>
```

The 

> T results = deserializer.Deserialize<T>(response);

 line in the WolframAplha.cs file threw this expection:

```
System.FormatException
  HResult=0x80131537
  Message=Input string was not in a correct format.
  Source=System.Private.CoreLib
  StackTrace:
   at System.Number.ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type)
   at System.String.System.IConvertible.ToSingle(IFormatProvider provider)
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
   at RestSharp.Deserializers.XmlDeserializer.Map(Object x, XElement root)
   at RestSharp.Deserializers.XmlDeserializer.Deserialize[T](IRestResponse response)
   at WolframAlphaNet.WolframAlpha.GetResponse[T](RestRequest request) in C:\Work\WolframAlphaNet\src\WolframAlphaNet\WolframAlpha.cs:line 390
   at WolframAlphaNet.WolframAlpha.Query(String query) in C:\Work\WolframAlphaNet\src\WolframAlphaNet\WolframAlpha.cs:line 374
   at FeladatGenerator.Models.Wolfram.Query(String problemAsText) in C:\Work\feladatgenerator\Source\FeladatGenerator\Models\Wolfram.cs:line 42
   at FeladatGenerator.Controllers.HomeController.Verify(Int32 problemId, String solutionInput) in C:\Work\feladatgenerator\Source\FeladatGenerator\Controllers\HomeController.cs:line 129
   at Microsoft.Extensions.Internal.ObjectMethodExecutor.Execute(Object target, Object[] parameters)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()

```

**Comment by andrasfuchs** [comment]
RestSharp deserializer has a problem the `timing` and `parsetiming` properties. 
I filed [a requests in the RestSharp repo](https://github.com/restsharp/RestSharp/issues/1452) to include more information about the misbehaving property, to make the debugging easier.

I'm not sure why the parsing of the `float` and `double` types throw exceptions, can this be a culture-specific problem?

I temporarily changed their types to string in the `QueryResult` class, and it works now, but I would rather have them as floats.

**Comment by Jaryn91** [comment]
I have the same exception when I run an example from repo (.NET Core 3.0).

**Comment by Genbox** [comment]
Honestly, RestSharp has been a pain in the butt and I've since moved to using pure HttpClient with Newtonsoft.Json though my own extension instead. I had no idea people were using my client for WolframAlpha, but now that I know, I can take a look at building a better version of this library.

**Comment by Genbox** [comment]
Just a quick update: I've rewritten the entire library from scratch. The issue here is easy to fix, but it has been 8 years since I wrote this library and the API has changed since then. Unfortunately, the Wolfram|Alpha API docs are still lacking, so it will take some time to get everything implemented and tested.

**Comment by andrasfuchs** [comment]
@Genbox Excellent, thank you!

**Comment by Genbox** [comment]
I've released version 3.0.0-alpha1 of the project. It is completely rewritten from scratch. The API now supports 2.6 (up from 2.0) of the Wolfram|Alpha API.

Note that the package has changed name: https://www.nuget.org/packages/Genbox.WolframAlpha/

Also, note that there are a lot of breaking changes. APIs have changed everywhere, but it should be pretty close to the old one. Please test it out and report any issues you may find.
